sockets_extension_test.go

```diff
@@ -0,0 +1,86 @@
+package wasi_test
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stealthrocket/wasi-go"
+)
+
+func TestInet4AddressMarshalJSON(t *testing.T) {
+	testMarshalJSON(t,
+		&wasi.Inet4Address{
+			Port: 4242,
+			Addr: [4]byte{192, 168, 0, 2},
+		},
+		`"192.168.0.2:4242"`,
+	)
+}
+
+func TestInet4AddressMarshalYAML(t *testing.T) {
+	testMarshalYAML(t,
+		&wasi.Inet4Address{
+			Port: 4242,
+			Addr: [4]byte{192, 168, 0, 2},
+		},
+		`192.168.0.2:4242`,
+	)
+}
+
+func TestInet6AddressMarshalJSON(t *testing.T) {
+	testMarshalJSON(t,
+		&wasi.Inet6Address{
+			Port: 4242,
+			Addr: [16]byte{
+				0x20, 0x01,
+				0x0d, 0xb8,
+				0x85, 0xa3,
+				0x08, 0xd3,
+				0x13, 0x19,
+				0x8a, 0x2e,
+				0x03, 0x70,
+				0x73, 0x48,
+			},
+		},
+		`"[2001:db8:85a3:8d3:1319:8a2e:370:7348]:4242"`,
+	)
+}
+
+func TestInet6AddressMarshalYAML(t *testing.T) {
+	testMarshalYAML(t,
+		&wasi.Inet6Address{
+			Port: 4242,
+			Addr: [16]byte{
+				0x20, 0x01,
+				0x0d, 0xb8,
+				0x85, 0xa3,
+				0x08, 0xd3,
+				0x13, 0x19,
+				0x8a, 0x2e,
+				0x03, 0x70,
+				0x73, 0x48,
+			},
+		},
+		`[2001:db8:85a3:8d3:1319:8a2e:370:7348]:4242`,
```

Comment:
Could this be interpreted as a YAML list rather than a string?

Reply:
That's what was happening before, now the MarshalYAML method ensures that we render a scalar value because we return a string.